sdk/tests/conformance/rendering/rendering-sampling-feedback-loop.html

```diff
@@ -62,82 +83,102 @@
 <script>
 "use strict";
 
-var wtu = WebGLTestUtils;
+const wtu = WebGLTestUtils;
 description("This test verifies the functionality of rendering to the same texture where it samples from.");
 
-var gl = wtu.create3DContext("example", undefined, 2);
+const gl = wtu.create3DContext("example");
 
-var width = 8;
-var height = 8;
-var tex0;
-var tex1;
-var fbo;
-var program;
-var positionLoc;
-var texCoordLoc;
+const width = 8;
+const height = 8;
+let tex;
+let fbo;
+let drawBuffers = null;
 
 if (!gl) {
     testFailed("WebGL context does not exist");
 } else {
     testPassed("WebGL context exists");
 
+    if (gl.drawBuffers) {
+        debug("Using webgl2.drawBuffers.");
+        drawBuffers = (x) => gl.drawBuffers(x);
+    } else {
+        const ext = gl.getExtension("WEBGL_draw_buffers");
+        if (ext) {
+            debug("Using WEBGL_draw_buffers.drawBuffersWEBGL.");
+            drawBuffers = (x) => ext.drawBuffersWEBGL(x);
+        }
+    }
+
     init();
 
     // The sampling texture is bound to COLOR_ATTACHMENT1 during resource allocation
     allocate_resource();
 
-    rendering_sampling_feedback_loop([gl.NONE, gl.COLOR_ATTACHMENT1], gl.INVALID_OPERATION);
-    rendering_sampling_feedback_loop([gl.COLOR_ATTACHMENT0, gl.COLOR_ATTACHMENT1], gl.INVALID_OPERATION);
-    rendering_sampling_feedback_loop([gl.COLOR_ATTACHMENT0, gl.NONE], gl.NO_ERROR);
+    rendering_sampling_feedback_loop(null);
+    if (drawBuffers) {
+        rendering_sampling_feedback_loop([gl.NONE]);
+    }
 }
 
 function init() {
-    program = wtu.setupProgram(gl, ['vshader', 'fshader'], ['aPosition', 'aTexCoord'], [0, 1]);
-    positionLoc = gl.getAttribLocation(program, "aPosition");
-    texCoordLoc = gl.getAttribLocation(program, "aTexCoord");
+    let shaders = ["vs100", "fs100"];
+    if (gl.texStorage2D) {
```

Comment:
Is this a version check? Don't we have a wtu thing for "is webgl2"? `getDefault3DContextVersion() === 2` I guess

Reply:
Yeah, but it's easier to do it this way, and it always works directly on whatever `gl` is passed in. TBH a gl.version() would be nice to inject when we create a context, but this works.